.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,43 @@
+---
+name: "CodeQL"
+
+on:
+  push:
+    branches: '*'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2
+        with:
+          persist-credentials: false
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@60168efe1c415ce0f5521ea06d5c2062adbeed1b  # v3.28.17
+        with:
+          languages: c-cpp
+
+      - name: Build project
+        run: |
+          sudo apt-get update -yy
+          sudo apt-get install -yy \
+            build-essential \
+            libgadu-dev \
+            libmeanwhile-dev \
+            libpurple-dev \
+            meson
+          meson setup build
+          ninja -C build
```

Comment:
Seems like we should be building the internal libraries instead of system ones.

Reply:
This adds 9 more issues from libgadu.

Reply:
Four of the new ones are fixed here, and the remaining 5 are pointing out that `memset` to 0 may be deleted if the buffer is unused.